sonic-net/sonic-buildimage
# Added makefile and dependencies for building sonic-stormond whl

#### Why I did it
This is a sister PR to the [stormond implementation](https://github.com/sonic-net/sonic-platform-daemons/pull/433).
[SONiC Storage Monitoring Daemon HLD](https://github.com/sonic-net/SONiC/pull/1481) and has following changes:

1. Adds the makefile and dependencies file so as to be able to build the sonic-stormond wheel. 
2. Changed vendor implementation of ssd_util to match sonic-platform-common change where names of sonic-platform-common/sonic_platform_base/sonic_ssd and its constituent files were changed.


##### Work item tracking
- Microsoft ADO **(number only)**: 17468992

#### How I did it
Added files needed to build the stormond wheel
Changed the import path to match the new folder structure in sonic-platform-common

#### How to verify it
Pull these changes and run:

` make -f Makefile.work BLDENV=bookworm KEEP_SLAVE_ON=no target/python-wheels/bookworm/sonic_stormond-1.0-py3-none-any.whl`

Verify that the wheel is successfully built:

```
assrinivasan@assrinivasan-dev-vm-1:/data/sonic/ssdmon/sonic-buildimage$ ls -l target/python-wheels/bookworm/sonic_stormond-1.0-py3-none-any.whl
-rw-r--r-- 1 assrinivasan assrinivasan 5977 May 22 18:24 target/python-wheels/bookworm/sonic_stormond-1.0-py3-none-any.whl

```
After Storage monitoring daemon and associated changes have been merged, pick up this change. Then run the ssdutil command as follows:

```
admin@str-s6100-acs-1:/usr/local/lib/python3.11/dist-packages$ cat ssdutil/* | grep -i "ssdutil"
SYSLOG_IDENTIFIER = "ssdutil"
        from ssd_util import SsdUtil
        log.log_warning("Platform specific SsdUtil module not found. Falling down to the generic implementation")
            **from sonic_platform_base.sonic_storage.ssd import SsdUtil**
            log.log_error("Failed to import default SsdUtil. Error: {}".format(str(e)), True)
    return SsdUtil(diskdev)
cat: def ssdutil():
ssdutil/__pycache__    ssdutil()

admin@str-s6100-acs-1:/usr/local/lib/python3.11/dist-packages$ sudo ssdutil
Device Model : InnoDisk Corp. - mSATA 3IE3
Health       : 99%
Temperature  : 30C
```




#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
